src/config.cpp

```
@@ -1,188 +1,187 @@
-				}
-				break;
-			case XMLReader::Inspected::EndTag:
+		// Write to userConfigs.json (never configs.json), so an FSMP update replacing configs.json can't wipe
+		// the user's settings. We write a clean serialization of only the fields we model, so a setting that
+		// a later FSMP version removes leaves no stale key behind. (Variant-only keys like enableCuda live in
+		// the shipped configs.json, which we never rewrite, so they are unaffected.)
+		const std::string merged = serializeConfigJson(readConfig());
+
+		// Atomic write: emit to a sibling temp file, then rename over the target so a crash mid-write can
+		// never leave a truncated userConfigs.json that would fail to parse on next launch.
+		std::error_code ec;
```

Clang (14.0.6)

[warning] variable name 'ec' is too short, expected at least 3 characters

(readability-identifier-length)